autoload/ale/autocmd.vim

```diff
@@ -1,77 +1,62 @@
 function! ale#autocmd#InitAuGroups() abort
-    " This value used to be a Boolean as a Number, and is now a String.
-    let l:text_changed = '' . g:ale_lint_on_text_changed
+    if g:ale_enabled
+        " This value used to be a Boolean as a Number, and is now a String.
+        let l:text_changed = '' . g:ale_lint_on_text_changed
 
-    augroup ALEPatternOptionsGroup
-        autocmd!
-        autocmd BufEnter,BufRead * call ale#pattern_options#SetOptions(str2nr(expand('<abuf>')))
-    augroup END
-
-    augroup ALERunOnTextChangedGroup
-        autocmd!
-        if g:ale_enabled
+        augroup ALETriggerGroup
```

Comment:
It might be best to leave the `autocmd!` lines in, if they don't cost too much to execute, in case a script is reloaded which runs this function again.